components/audioinput/audio_input.go

```diff
@@ -84,13 +84,33 @@ func WrapWithReconfigurable(r interface{}, name resource.Name) (resource.Reconfi
 	if reconfigurable, ok := i.(*reconfigurableAudioInput); ok {
 		return reconfigurable, nil
 	}
+	reconfigurable := newReconfigurable(i, name)
+
+	if mon, ok := i.(LivenessMonitor); ok {
+		mon.Monitor(func() {
+			reconfigurable.reconfigure(i)
+		})
+	}
+
+	return reconfigurable, nil
+}
+
+func newReconfigurable(i AudioInput, name resource.Name) *reconfigurableAudioInput {
 	cancelCtx, cancel := context.WithCancel(context.Background())
 	return &reconfigurableAudioInput{
 		name:      name,
 		actual:    i,
 		cancelCtx: cancelCtx,
 		cancel:    cancel,
-	}, nil
+	}
+}
+
+// A LivenessMonitor is responsible for monitoring the liveness of an audio input. An example
+// is connectivity. Since the model itself knows best about how to maintain this state,
+// the reconfigurable offers a safe way to notify if a state needs to be reset due
+// to some exceptional event (like a reconnect).
+type LivenessMonitor interface {
```

Comment:
if we like how camera works, I'll file a ticket to implement this for microphone.